persist-cli/src/main/java/io/ballerina/persist/cmd/Generate.java

```diff
@@ -133,210 +112,103 @@ public void execute() {
+                    errStream.println("ERROR: invalid module name : '" + moduleName + "' :\n" +
+                            "maximum length of module name is 256 characters");
                     return;
                 }
-                generateDataTypes(entityModule, generatedSourceDirPath);
-                generateClientBalFile(entityModule, generatedSourceDirPath, dataStore);
+                generatedSourceDirPath = generatedSourceDirPath.resolve(moduleName);
+            }
+            dataStore = ballerinaTomlConfig.get("datastore").trim();
+            if (!PersistToolsConstants.SUPPORTED_DB_PROVIDERS.contains(dataStore)) {
+                errStream.printf("ERROR: the persist layer supports one of data stores: %s" +
+                                ". but found '%s' datasource.%n",
+                        Arrays.toString(PersistToolsConstants.SUPPORTED_DB_PROVIDERS.toArray()),
+                        ballerinaTomlConfig.get("datastore").trim());
+                return;
+            }
+        } catch (BalException e) {
+            errStream.printf("ERROR: failed to generate types and client for the definition file(%s). %s%n",
+                    "Ballerina.toml", e.getMessage());
+            return;
+        }
 
-            } catch (BalException e) {
-                errStream.printf("ERROR: failed to generate types and client for the definition file(%s). %s%n",
-                        file.getFileName(), e.getMessage());
+        if (!Files.exists(generatedSourceDirPath)) {
+            try {
+                Files.createDirectories(generatedSourceDirPath.toAbsolutePath());
+            } catch (IOException e) {
+                errStream.println("ERROR: failed to create the generated directory. " + e.getMessage());
                 return;
             }
-            if (dataStore.equals(PersistToolsConstants.SupportDataSources.MYSQL_DB)) {
-                try {
-                    ArrayList<Entity> entityArray = new ArrayList<>(entityModule.getEntityMap().values());
-                    String[] sqlScripts = SqlScriptGenerationUtils.generateSqlScript(entityArray);
-                    SqlScriptGenerationUtils.writeScriptFile(entityModule.getModuleName(), sqlScripts,
-                            generatedSourceDirPath);
-                } catch (BalException e) {
-                    errStream.printf("ERROR: failed to generate SQL schema for the definition file(%s). %s%n",
-                            file.getFileName(), e.getMessage());
-                }
+        }
+        SourceGenerator sourceCreator = new SourceGenerator(sourcePath, generatedSourceDirPath, packageNameInToml,
+                entityModule);
+        if (dataStore.equals(PersistToolsConstants.SupportDataSources.MYSQL_DB)) {
+            try {
+                sourceCreator.createDbSources();
+            } catch (BalException e) {
+                errStream.printf(e.getMessage());
+                return;
```

Comment:
Let's move this one also to the function. Follow the convention we use to print the Error messages.

Reply:
Updated the error message. Which function should this logic be moved into?